core/chains/solana/chain.go

```diff
@@ -123,33 +128,42 @@ func (c *chain) getClient() (solanaclient.ReaderWriter, error) {
 
 // verifiedClient returns a client for node or an error if the chain id does not match.
 func (c *chain) verifiedClient(node db.Node) (solanaclient.ReaderWriter, error) {
-	// create client
-	client, err := solanaclient.NewClient(node.SolanaURL, c.cfg, DefaultRequestTimeout, c.lggr.Named("Client-"+node.Name))
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to create client")
-	}
-
-	// if endpoint has not been checked, fetch chainID
 	url := node.SolanaURL
-	c.chainIDLock.RLock()
-	id, exists := c.clientChainID[url]
-	c.chainIDLock.RUnlock()
+	var err error
+
+	// check if cached client exists
+	c.clientLock.RLock()
+	client, exists := c.clientCache[url]
+	c.clientLock.RUnlock()
+
 	if !exists {
-		id, err = client.ChainID()
+		// create client
+		client.rw, err = solanaclient.NewClient(url, c.cfg, DefaultRequestTimeout, c.lggr.Named("Client-"+node.Name))
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create client")
+		}
+
+		client.id, err = client.rw.ChainID()
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to fetch ChainID in checkClient")
 		}
-		c.chainIDLock.Lock()
-		c.clientChainID[url] = id
-		c.chainIDLock.Unlock()
 	}
 
 	// check chainID matches expected chainID
 	expectedID := strings.ToLower(c.id)
-	if id != expectedID {
-		return nil, errors.Errorf("client returned mismatched chain id (expected: %s, got: %s): %s", expectedID, id, url)
+	if client.id != expectedID {
```

Comment:
So we do expect `client.id` to be lowercase, but where is that enforced?

Reply:
the call from the `ChainID()` method call on the client will always return lowercase: https://github.com/smartcontractkit/chainlink-solana/blob/ba505cd5055f1d96b8efc1000234a2f851aa2819/pkg/solana/client/client.go#L106